Remotion/Data/DomainObjects.UnitTests/Persistence/Rdbms/Parameters/SerializedObjectIDDataParameterDefinitionTest.cs

```diff
@@ -0,0 +1,147 @@
+    var storageTypeInformation = StorageTypeInformationObjectMother.CreateIntStorageTypeInformation();
+
+    Assert.That(
+        () => new SerializedObjectIDDataParameterDefinition(storageTypeInformation),
+        Throws.InstanceOf<ArgumentException>().With.ArgumentExceptionMessageEqualTo(
+            $"Parameter 'storageTypeInformation.DotNetType' is a '{typeof(int)}', which cannot be assigned to type '{typeof(string)}'.",
+            "storageTypeInformation.DotNetType"));
+  }
+
+  [Test]
+  public void GetParameterValue ()
+  {
+    var storageTypeInformation = StorageTypeInformationObjectMother.CreateStorageTypeInformation(storageDbType: DbType.AnsiString, storageTypeLength: 255);
+
+    var objectID = DomainObjectIDs.Order1;
+    var expectedValue = ObjectIDStringSerializer.Instance.Serialize(objectID);
+
+    var serializedObjectIDDataParameterDefinition = new SerializedObjectIDDataParameterDefinition(storageTypeInformation);
+
+    var result = serializedObjectIDDataParameterDefinition.GetParameterValue(objectID);
+
+    Assert.That(result, Is.EqualTo(expectedValue));
+  }
+
+  [Test]
+  public void GetParameterValue_WithNull_ReturnsDBNull ()
+  {
+    var storageTypeInformation = StorageTypeInformationObjectMother.CreateStorageTypeInformation();
+
+    var serializedObjectIDDataParameterDefinition = new SerializedObjectIDDataParameterDefinition(storageTypeInformation);
+
+    Assert.That(serializedObjectIDDataParameterDefinition.GetParameterValue(null), Is.EqualTo(DBNull.Value));
+  }
+
+  [Test]
+  public void GetParameterValue_WithTypeMismatchedValue_ThrowsArgumentException ()
+  {
+    var dummyValue = "Not an ObjectID";
+    var storageTypeInformation = StorageTypeInformationObjectMother.CreateStorageTypeInformation();
+
+    var serializedObjectIDDataParameterDefinition = new SerializedObjectIDDataParameterDefinition(storageTypeInformation);
+
+    Assert.That(
+        () => serializedObjectIDDataParameterDefinition.GetParameterValue(dummyValue),
+        Throws.InstanceOf<ArgumentException>().With.ArgumentExceptionMessageEqualTo(
+            $"Parameter 'value' has type '{typeof(string)}' when type '{typeof(ObjectID)}' was expected.",
+            "value"));
+  }
+
+  [Test]
+  public void CreateDataParameter ()
+  {
+    var commandStub = new Mock<IDbCommand>();
+    var dataParameterMock = new Mock<IDbDataParameter>();
+
+    var storageTypeInformation = StorageTypeInformationObjectMother.CreateStorageTypeInformation(storageDbType: DbType.AnsiString, storageTypeLength: 255);
+
+    var objectID = DomainObjectIDs.Order1;
+    var testValue = ObjectIDStringSerializer.Instance.Serialize(objectID);
+
+    commandStub.Setup(mock => mock.CreateParameter()).Returns(dataParameterMock.Object);
+
+    dataParameterMock.SetupProperty(mock => mock.ParameterName);
+    dataParameterMock.SetupProperty(mock => mock.DbType);
+    dataParameterMock.SetupProperty(mock => mock.Value);
+    dataParameterMock.SetupProperty(mock => mock.Size);
```

Comment:
stylistic input: for new code, we use `_` instead of `mock` or `stub` in expressions to make the varable less prominent.